xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-ui/src/main/java/org/xwiki/test/ui/WcagContext.java

```diff
@@ -0,0 +1,412 @@
+        if (this.isNotCached(url, className)) {
+            /* Avoid duplicate entries in the cache. */
+            this.wcagValidationCache.putIfAbsent(url, new ArrayList<>());
+            this.wcagValidationCache.get(url).add(className);
+        }
+    }
+
+    /**
+     * @param time The amount of time to count up, in ms.
+     */
+    public void addWcagTime(long time)
+    {
+        this.wcagTimer += time;
+    }
+
+    /**
+     * @return the accumulated time spent validating accessibility.
+     */
+    public long getWcagTime()
+    {
+        return this.wcagTimer;
+    }
+
+    /**
+     * @param wcag validation enabled setup parameter to use in the test suite.
+     */
+    public void setWcagEnabled(boolean wcag)
+    {
+        this.wcagEnabled = wcag;
+    }
+
+    /**
+     * @return the state of wcag validation
+     */
+    public boolean isWcagEnabled()
+    {
+        return this.wcagEnabled;
+    }
+
+    /**
+     * @return any of the results about fails stored is not empty
+     */
+    public boolean hasWCAGFails()
+    {
+        for (WCAGTestResults results : this.wcagResults) {
+            if (!results.isFailEmpty()) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * @return any of the results about warnings stored is not empty
+     */
+    public boolean hasWCAGWarnings()
+    {
+        for (WCAGTestResults results : this.wcagResults) {
+            if (!results.isWarnEmpty()) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * @return a readable report of wcag fails found so far.
+     */
+    public String buildFailsReport()
+    {
+        StringBuilder mergedReport = new StringBuilder();
+        boolean failsEmpty = true;
+        for (WCAGTestResults result : this.wcagResults) {
+            if (!result.isFailEmpty()) {
+                if (failsEmpty) {
+                    mergedReport.append(String.format("WCAG fails in the test class %s:", getTestClassName()));
+                    mergedReport.append(System.lineSeparator());
+                    failsEmpty = false;
+                }
+                mergedReport.append(result.getFailReport());
+                mergedReport.append(System.lineSeparator());
+                mergedReport.append(System.lineSeparator());
+            }
+        }
+        return mergedReport.toString();
+    }
+
+    /**
+     * @return a readable report of wcag violations (non failing) found so far.
+     */
+    public String buildWarningsReport()
+    {
+        StringBuilder mergedReport = new StringBuilder();
+        boolean warningsEmpty = true;
+        for (WCAGTestResults result : this.wcagResults) {
+            if (!result.isWarnEmpty()) {
+                if (warningsEmpty) {
+                    mergedReport.append(String.format("WCAG warnings in the test class %s:", getTestClassName()));
```

Comment:
Not using our logging best practice of using `[...]`. See https://dev.xwiki.org/xwiki/bin/view/Community/CodeStyle/JavaCodeStyle/#HLoggingBestPractices